tests/support/support.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { RadiclePeer } from "./peerManager";
+
+import { fileURLToPath } from "url";
+import { dirname } from "path";
+import * as Path from "node:path";
+import * as Fs from "node:fs/promises";
+
+import { gitOptions } from "@tests/support/fixtures";
+
+const filename = fileURLToPath(import.meta.url);
+const dir = dirname(filename);
+
+// Create a project using uphe rad CLI.
+export async function createProject(
+  peer: RadiclePeer,
+  name: string,
+  description = "",
+): Promise<{ rid: string }> {
+  const cwd = Path.join(peer.checkoutPath, name);
+  await Fs.mkdir(cwd, { recursive: true });
+  await peer.git(["init", "--initial-branch", "master"], {
+    cwd,
+    env: gitOptions["alice"],
+  });
+  await peer.git(["commit", "--allow-empty", "--message", "initial commit"], {
+    cwd,
+    env: gitOptions["alice"],
+  });
+  await peer.rad(
+    [
+      "init",
+      "--name",
+      name,
+      "--default-branch",
+      "master",
+      "--description",
+      description,
+      "--announce",
+    ],
+    { cwd },
+  );
+
+  const { stdout: rid } = await peer.rad(["inspect"], { cwd });
+
+  return { rid };
+}
+
+// Create a project from the platinum fixture using the rad CLI.
+export async function createProjectFromSourceBrowsingFixture(
+  peer: RadiclePeer,
+  projectName = "source-browsing",
+): Promise<{
+  rid: string;
+}> {
+  await peer.git(["clone", Path.join(dir, "..", "tmp", projectName)], {
+    cwd: peer.checkoutPath,
+  });
+  const cwd = Path.join(peer.checkoutPath, projectName);
+
+  await peer.git(["checkout", "main"], { cwd });
+  await peer.rad(
+    [
+      "init",
+      "--name",
+      projectName,
+      "--default-branch",
+      "main",
+      "--description",
+      "Git repository for source browsing tests",
+      "--announce",
+    ],
+    { cwd },
+  );
+
+  await peer.git(["checkout", "feature/branch"], { cwd });
+  await peer.git(["push", "rad"], { cwd });
+  await peer.git(["checkout", "orphaned-branch"], { cwd });
+  await peer.git(["push", "rad"], { cwd });
+
+  const { stdout: rid } = await peer.rad(["inspect"], { cwd });
+
+  return {
+    rid,
+  };
+}
+
+export async function modifySourceBrowsingFixture(
+  peer: RadiclePeer,
+  projectName = "source-browsing",
+) {
+  const cwd = Path.join(peer.checkoutPath, projectName);
+  // @ts-expect-error Some weirdness with the type of `spawn`.
+  await peer
+    .spawn("echo", ["Updated readme"])
+    .pipeStdout(Path.join(cwd, "README.md"));
```

Comment:
I’m not sure this does what it is supposed to. Instead you can just change the file using [`fsPromises.writeFile()`](https://nodejs.org/api/fs.html#fspromiseswritefilefile-data-options).